orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/ExecutionLauncher.java

```diff
@@ -64,6 +75,36 @@ public T start(T execution) throws Exception {
     return execution;
   }
 
+  protected T handleStartupFailure(T execution, Throwable failure) {
+    final String canceledBy = "system";
+    final String reason = "Failed on startup: " + failure.getMessage();
+    final ExecutionStatus status = ExecutionStatus.TERMINAL;
+    executionRepository.updateStatus(execution.getId(), status);
```

Comment:
cancel only changes the status to CANCELED if it is in NOT_STARTED (I think..). Resulting pipeline ends up in TERMINAL when this fires